worlds/adventure/__init__.py

```diff
@@ -0,0 +1,355 @@
+                self.dragon_speed_reducer_info[item_table[item_name].id] = [speed_with_one, min_speed]
+                self.created_items += 2
+            elif maximum_items >= 1:
+                self.multiworld.itempool.append(self.create_item(item_name))
+                self.dragon_speed_reducer_info[item_table[item_name].id] = [min_speed]
+                self.created_items += 1
+
+    def create_regions(self) -> None:
+        create_regions(self.multiworld, self.player, self.dragon_rooms)
+
+    def set_rules(self) -> None:
+        set_rules(self)
+
+    def generate_basic(self) -> None:
+        self.multiworld.get_location("Chalice Home", self.player).place_locked_item(
+            self.create_event("Victory", ItemClassification.progression))
+        self.multiworld.completion_condition[self.player] = lambda state: state.has("Victory", self.player)
+
+    def pre_fill(self):
+        # Place empty items in filler locations here, to limit
+        # the number of exported empty items and the density of stuff in overworld.
+        overworld = self.multiworld.get_region("Overworld", self.player)
+        overworld_locations_copy = overworld.locations.copy()
+        all_locations = self.multiworld.get_locations(self.player)
+        total_location_count = len(all_locations)
+        locations_copy = all_locations.copy()
+        for loc in all_locations:
+            if loc.item is not None or loc.progress_type != LocationProgressType.DEFAULT:
+                locations_copy.remove(loc)
+                if loc in overworld_locations_copy:
+                    overworld_locations_copy.remove(loc)
+
+        # unfilled_locations = len(locations_copy)
+        # filled_locations = len(overworld.locations) - unfilled_locations
+        force_empty_item_count = (total_location_count - self.created_items)
+        nothing_items = list(filter(lambda i: i.name == "nothing", self.multiworld.itempool))
+        # guarantee at least one overworld location, so we can for sure get a key somewhere
+        saved_overworld_loc = self.multiworld.random.choice(overworld_locations_copy)
+        locations_copy.remove(saved_overworld_loc)
+        overworld_locations_copy.remove(saved_overworld_loc)
+        while force_empty_item_count > 0 and len(locations_copy) > 0 and len(nothing_items) > 0:
+            force_empty_item_count -= 1
+            # prefer somewhat to thin out the overworld.
+            if len(overworld_locations_copy) > 0 and self.multiworld.random.randint(0, 10) < 4:
+                loc = self.multiworld.random.choice(overworld_locations_copy)
+            else:
+                loc = self.multiworld.random.choice(locations_copy)
+            item = nothing_items.pop()
+            loc.place_locked_item(item)
+            locations_copy.remove(loc)
+            if loc in overworld_locations_copy:
+                overworld_locations_copy.remove(loc)
+            self.multiworld.itempool.remove(item)
```

Comment:
in large multiworlds, this is a very slow operation

Reply:
Well, I'm not adding or removing those from the itempool anymore.  I think the player experience of not generating the locations is worse, so I'd rather not do that, but if that's the preferred AP way to do it, I can change back over and look into how to make sure I create any locations specified in plandos.  
Could you let me know?  Thanks.